C-Unit1-assignments/src/Assignment_01.cpp

```diff
@@ -0,0 +1,204 @@
+/*
+Note:
+1. Read instructions for each function carefully.
+2. Feel free to create new functions if needed. Give good names!
+3. Check the reference links when ever given.
+4. After solving the question and making corresponding test cases pass
+   - reflect on where is the gap in your understanding, while solving the problem.
+   - if you have to create a similar problem, how would you do/tweek it?
+   - when your friends seek help, instead of explaining your logic,
+     give hints to solve. Always.
+*/
+
+//
+// butterfly number for the given integer
+//
+// e.g:
+// 20  => "2002"
+// -2  => "-22-"
+//
+// To get butterfly number for the given number 20,
+// take its reversal 02 and join them to form "2002".
+//
+
+
+
+void reverse(char butterfly[], char temp[]) {
+	int i;
+	for (i = 0;butterfly[i] != '\0';i++);
+	int j = 0;
+	while (i) {
+		temp[j++] = butterfly[--i];
+	}temp[j] = '\0';
+}
+void butterflyNumber(int n, char butterfly[23]) {
+	long i = 0;
+	long x = n;
+	if (n == 0) {
+		butterfly[0] = butterfly[1] = '0';
+		butterfly[2] = '\0';
+		return;
+	}
+	long l = 0;
+	if (n < 0) {
+		butterfly[0] = '-';
+		i++;//l++;
+	}
+	//printf("%d ",n);
+	if (n<0)
+		x = -n;
+	else x = n;
+	long r = 0;
+	while (x) {
+		r *= 10;
+		r += x % 10;
+		//printf("%d\n",r);
+		x = x / 10;l++;
+		//printf("%d ",x);
+	}
+	long t = l;//printf("%d ",l);
+
```

Comment:
Can you make this modular. Break it down into some functions and reuse them. Try not to have more than 10 lines in a function from now onwards.